chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -385,60 +390,127 @@ out:
 // historicalConfDetails looks up whether a transaction is already included in a
 // block in the active chain and, if so, returns details about the confirmation.
 func (b *BitcoindNotifier) historicalConfDetails(txid *chainhash.Hash,
+	heightHint, currentHeight uint32) (*chainntnfs.TxConfirmation, error) {
+
+	// First, we'll attempt to retrieve the transaction details using the
+	// backend node's transaction index.
+	txConf, err := b.confDetailsFromTxIndex(txid)
+	if err != nil {
```

Comment:
Noticed this while addressing your comments: maybe we should proceed to use the `heightHint` if this fails, rather than returning the error?

Reply:
With the new code, the nature of the error is _likely_ critical. We'll need to properly test with both `bitcoind` and `btcd` to ensure that our logic catches the "no tx index" error properly.